.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   
+  pull_request:  
+
+jobs:
+  build:
```

Comment:
You'd also have to set `RUSTFLAGS` to fail `cargo hack check` / `cargo clippy` in case of warnings:

```yaml
jobs:
  build:
    runs-on: ubuntu-latest

    env:
      RUSTFLAGS: "-D warnings"
```